packages/framework/src/Framework/Features/Publications/PublicationService.php

```diff
@@ -51,29 +51,40 @@ public static function getPublicationsForPubType(PublicationType $pubType): Coll
     /**
      * Return all media items for a given publication type.
      */
-    public static function getMediaForPubType(PublicationType $pubType): Collection
+    public static function getMediaForPubType(PublicationType $pubType, bool $reload = true): Collection
     {
-        return Collection::create(static::getMediaFiles($pubType->getDirectory()))->map(function (string $file): string {
-            return Hyde::pathToRelative($file);
-        });
+        static $mediaFiles = null;
+        if (!$mediaFiles || $reload) {
+            $mediaFiles =  Collection::create(static::getMediaFiles($pubType->getDirectory()))->map(function (string $file): string {
+                return Hyde::pathToRelative($file);
+            });
+        }
+
+        return $mediaFiles;
     }
 
     /**
      * Get all available tags.
      *
+     * @param bool  $reload Reload the tags from the filesystem
      * @return \Rgasch\Collection\Collection
      *
      * @throws \Safe\Exceptions\FilesystemException
      * @throws \Safe\Exceptions\JsonException
      */
-    public static function getAllTags(): Collection
+    public static function getAllTags(bool $reload=true): Collection
     {
         $filename = Hyde::pathToRelative('tags.json');
         if (! file_exists($filename)) {
             return Collection::create();
         }
 
-        return Collection::create(json_decode(file_get_contents($filename), true))->sortKeys();
+        static $tags = null;
```

Comment:
What's the purpose of the static variable here? Is it just to cache the discovered tags? Or does this actually make so the data is reloaded automatically when using the returned variable?

I haven't really used (non-class) static variables before.